src/main/kotlin/org/elm/lang/core/ElmTextUtils.kt

```diff
@@ -0,0 +1,15 @@
+package org.elm.lang.core
+
+/**
+ * Convert a string so that its first character is guaranteed to be lowercase.
+ * This is necessary in some parts of Elm's syntax (e.g. a function parameter).
+ *
+ * If the receiver consists of all uppercase letters, the entire thing will be made
+ * lowercase (because "uuid" is a far more sensible transformation of "UUID" than "uUID").
+ */
+fun String.toElmLowerId(): String =
+        when {
+            isEmpty() -> ""
+            all { it.isUpperCase() } -> toLowerCase()
```

Comment:
This fixes a long-standing bug when generating a func decl from a type annotation. If you had `f : UUID -> String` and you invoked the 'make declaration' intention, it would generate `f uUID = ...`, which is lame. Now it will generate `f uuid = ...`